src/connection.h

```diff
@@ -255,8 +256,18 @@ static inline int connSetWriteHandlerWithBarrier(connection *conn, ConnectionCal
     return conn->type->set_write_handler(conn, func, barrier);
 }
 
-static inline void connShutdown(connection *conn) {
-    conn->type->shutdown(conn);
+static inline void connShutdown(connection *conn, int force) {
+    if (force) {
+        /*
+         * When the 'force' flag is set, we perform a forceful shutdown of the
+         * connection, regardless of whether it is currently blocked or not.
+         * This is useful in situations where we need to terminate the connection
+         * immediately, without waiting for any pending operations to complete.
+         */
+        shutdown(conn->fd, SHUT_RDWR);
+    } else {
+        conn->type->shutdown(conn);
```

Comment:
The connection abstraction shouldn't know there is an underlying socket it can shutdown. The implementation should be in both the underlying TLS implementation and socket connection. 
```suggestion
        conn->type->shutdown(conn, force);
```

Reply:
Ok, I'll try to unblocked the connection before shutting it down, this way we can use the customized connection shutdown. 
@madolson Can we please run the daily tests WF on that?